libs/src/api/Track.ts

```diff
@@ -709,10 +763,33 @@ export class Track extends Base {
    * Marks a tracks as deleted
    * @param trackId
    */
-  async deleteTrack(trackId: number) {
-    return await this.contracts.TrackFactoryClient.deleteTrack(trackId)
+  async deleteTrack(trackId: number, useEntityManager: boolean) {
+    if (useEntityManager) {
+      const ownerId = this.userStateManager.getCurrentUserId()
+
+      if (!ownerId) throw new Error('No users loaded for this wallet')
+
+      return await this.contracts.EntityManagerClient!.manageEntity(
+        ownerId,
+        EntityManagerClient.EntityType.TRACK,
+        trackId,
+        EntityManagerClient.Action.DELETE,
+        ''
+      )
+    } else {
+      return await this.contracts.TrackFactoryClient.deleteTrack(trackId)
+    }
   }
 
+  // Minimum track ID, intentionally higher than legacy track ID range
+  static MIN_TRACK_ID = 1000000
+
+  // Maximum track ID, reflects postgres max integer value
+  static MAX_TRACK_ID = 2147483647
```

Comment:
Can you confirm that 2147483647 is not too big to cause any sort of collision with our hash id stuff? My best guess is we have 5 letters `BqN19` so 62 ^ 5 which is less than 2 billion

Reply:
i think it works actually! 5 is just min length and 2147483647 encodes to a 7 letter string

> There are no collisions because the method is based on integer to hex conversion. As long as you don't change constructor arguments midway, the generated output will stay unique to your salt.

Reply:
AMAZING